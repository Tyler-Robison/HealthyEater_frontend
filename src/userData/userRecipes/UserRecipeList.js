import React, { useContext } from "react";
import UserRecipe from "./UserRecipe";
import GlobalContext from "../../context/GlobalContext";
import { Link } from "react-router-dom";
import './UserRecipeList.css'

/** UserRecipeList displays all recipes saved by currentUser
 * 
 * recipes deleted here will cascade to the mealplanner*/
const UserRecipeList = () => {
    const { currentUser } = useContext(GlobalContext)

    if (currentUser.recipes.length === 0) return (
        <p className="no-recipe-para">You have no saved recipes, go to <Link to={`/findrecipes`} className='RecipeList-para'><b>Find Recipes</b></Link></p>
    )

    return (
        <div className="UserRecipeList">
         
            <div className="UserRecipeList-div">
            <h3 className="UserRecipeList-header">{currentUser.username}'s saved recipes</h3>
                <div className="saved-recipe-container">
                    {currentUser.recipes.map(recipe => {
                        return <UserRecipe key={recipe.recipe_id} recipe={recipe} />
                    })}
                </div>
            </div>
        </div>
    )
}

export default UserRecipeList;