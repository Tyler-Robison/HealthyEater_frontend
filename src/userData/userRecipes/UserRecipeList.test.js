import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import UserRecipeList from './UserRecipeList';
import { MemoryRouter } from "react-router-dom";
import ContextProvider from '../../testContext';
import mockedAxios from 'axios';



it("renders without crashing", function () {
    <MemoryRouter>
        <UserRecipeList />
    </MemoryRouter>
});

it("matches snapshot", function () {

    const container = render(
        <MemoryRouter>
            <ContextProvider >
                <UserRecipeList />
            </ContextProvider>
        </MemoryRouter>
    );
    expect(container.asFragment()).toMatchSnapshot();
});

it("correctly displays users's saved recipes", function () {

    render(
        <MemoryRouter>
            <ContextProvider >
                <UserRecipeList />
            </ContextProvider>
        </MemoryRouter>
    );

    expect(screen.getByText('bacon')).toBeInTheDocument();
    expect(screen.getByText('(20 points)')).toBeInTheDocument();
    expect(screen.getByText('eggs')).toBeInTheDocument();
    expect(screen.getByText('(10 points)')).toBeInTheDocument();
});

it("Can remove user saved recipes", async function () {

    const mockData = {
        data: {
            deletedRecipe: { recipe_id: 1111 }
        }
    }

    mockedAxios.delete.mockResolvedValueOnce(mockData);

    render(
        <MemoryRouter>
            <ContextProvider >
                <UserRecipeList />
            </ContextProvider>
        </MemoryRouter>
    );

    expect(screen.getByText('bacon')).toBeInTheDocument();
    expect(screen.getByText('(20 points)')).toBeInTheDocument();
    expect(screen.getByText('eggs')).toBeInTheDocument();
    expect(screen.getByText('(10 points)')).toBeInTheDocument();

    const removeBaconBtn = screen.getByTestId(1111)
    fireEvent.click(removeBaconBtn)

    await waitFor(() => {
        expect( mockedAxios.delete.mockResolvedValueOnce(mockData)).toHaveBeenCalledTimes(1)
        expect(screen.queryByText('bacon')).not.toBeInTheDocument();
        expect(screen.queryByText('(20 points)')).not.toBeInTheDocument();
        expect(screen.getByText('eggs')).toBeInTheDocument();
        expect(screen.getByText('(10 points)')).toBeInTheDocument();
    })

});