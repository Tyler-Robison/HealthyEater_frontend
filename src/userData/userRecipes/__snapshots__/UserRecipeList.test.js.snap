// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="UserRecipeList"
  >
    <div
      class="UserRecipeList-div"
    >
      <h3
        class="UserRecipeList-header"
      >
        Tyler's saved recipes
      </h3>
      <div>
        <div>
          <a
            class="RecipeList-para"
            href="/1111/detail"
          >
            <b>
              bacon
            </b>
          </a>
           (20 points)
          <button
            class="general-btn-red small-btn"
            style="margin-left: 0.4rem;"
          >
            X
          </button>
        </div>
      </div>
      <div>
        <div>
          <a
            class="RecipeList-para"
            href="/2222/detail"
          >
            <b>
              eggs
            </b>
          </a>
           (10 points)
          <button
            class="general-btn-red small-btn"
            style="margin-left: 0.4rem;"
          >
            X
          </button>
        </div>
      </div>
      <div>
        <div>
          <a
            class="RecipeList-para"
            href="/3333/detail"
          >
            <b>
              salad
            </b>
          </a>
           (5 points)
          <button
            class="general-btn-red small-btn"
            style="margin-left: 0.4rem;"
          >
            X
          </button>
        </div>
      </div>
      <div>
        <div>
          <a
            class="RecipeList-para"
            href="/4444/detail"
          >
            <b>
              bread
            </b>
          </a>
           (7 points)
          <button
            class="general-btn-red small-btn"
            style="margin-left: 0.4rem;"
          >
            X
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
